code/datums/status_effects/xeno_buffs.dm

```diff
@@ -840,3 +840,32 @@
 	buff_owner.soft_armor = buff_owner.soft_armor.detachArmor(armor_modifier)
 	armor_modifier = null
 	return ..()
+
+// ***************************************
+// *********** Queen Screeches
+// ***************************************
+/datum/status_effect/frenzy_screech
+	id = "frenzy_screech"
+	status_type = STATUS_EFFECT_REFRESH
+	alert_type = null
+	var/mob/living/carbon/xenomorph/buff_owner
+	var/modifier
+
+/datum/status_effect/frenzy_screech/on_creation(mob/living/new_owner, set_duration, damage_modifier)
+	duration = set_duration
+	owner = new_owner
+	modifier = damage_modifier
+	return ..()
+
+/datum/status_effect/frenzy_screech/on_apply()
+	if(!isxeno(owner))
+		return FALSE
+	buff_owner = owner
```

Comment:
`/datum/status_effect/blessing` had it like that and since `/datum/status_effect/blessing/fury` basically looked like what I wanted, I saw no issue in copying it and it's parent.